### DISCUSSION TITLE
How to make the publication type visible and filterable

### DISCUSSION THREAD

**Discussion by imnanxu** [greenelab/lab-website-template#268] [Q&A] [answered]
### Checks

- [X] I have searched **[the docs](https://greene-lab.gitbook.io/lab-website-template-docs)**, [existing issues](https://github.com/greenelab/lab-website-template/issues), and [existing discussions](https://github.com/greenelab/lab-website-template/discussions) for answers first.

### Link to your website repo

https://inspirelab-site.github.io

### Version of Lab Website Template you are using

1.2.2

### Description

Hi, I wonder how to make the style of publication list exactly as shown in https://manubot.org/catalog/. Specifically, I want to:
- display the publication type (e.g., journal, conference, etc.) by the right side, which can be filtered at the top
- disable the section titles for publication years
- ensure all paper images are a fixed size

The current website template doesn't have these features by default.

In addition, I also have the issue with the icon and favicon. 
- The favicon still displays the website lab template even though I have replaced all files in ./images.
- The icon gets squeezed when narrowing the window size or browsing on iPhone/iPad. 
<img width="574" alt="Screenshot 2024-07-24 at 10 57 54 AM" src="https://github.com/user-attachments/assets/a3a46f21-b9c7-4614-a1a3-84a4183e1d56">

Thanks,
N

**Comment by vincerubinetti** [comment, marked as answer]
Hi,

In the future, please provide a link to your repo (github source code, not a link to the website itself). Had to dig to find this:
https://github.com/inspirelab-site/inspirelab-site.github.io

I'll answer your questions in order:

---

> display the publication type (e.g., journal, conference, etc.) by the right side, which can be filtered at the top

As you seem to already be doing, you can add a manual `type` to each citation, but that only [specifies the faded icon to show](https://greene-lab.gitbook.io/lab-website-template-docs/basics/citations#rich-details) as a hint to the user. Instead, I would recommend [tagging](https://greene-lab.gitbook.io/lab-website-template-docs/basics/citations#rich-details) each citation with `paper`, `conference`, etc. Then you can use the [tags component](https://greene-lab.gitbook.io/lab-website-template-docs/basics/components/tags) at the top of the page with manual tags (e.g. `tags="paper, conference, etc"`) which the user can then click on to filter.

If this is not desirable...

I would likely not build in a "filter by publication type" feature into the template because it too inflexible and not in demand enough. I would rather try to implement a "filter by any field" feature, but that would be difficult to accomplish because the filtering happens at in the browser (at run-time, not at compile-time) after those fields have already been rendered as html in various ways that is brittle to track. You can create an issue for a feature request for this if you want, but I'd say it's unlikely to be implemented as that's already what the tags feature is designed to do.

> disable the section titles for publication years

You can go into `/_includes/list.html` and remove the `{{--}}<h3 id="{{ year.name }}">{{ year.name }}</h3>` line which shows the year. 

Related to the discussion above, we can't add template "options" for low-level things like this. As this is a template and not a package, you're meant to go in and edit things yourself, except for the most commonly needed configurations. We need to do this for practicality (we don't want to end up with 100 different small options/flags) and to avoid [anti-patterns](https://en.wikipedia.org/wiki/Inner-platform_effect).

> ensure all paper images are a fixed size

You'll have to clarify what you mean by this. 

Perhaps you want to change the image's [object fit](https://developer.mozilla.org/en-US/docs/Web/CSS/object-fit) to "cover", instead of "contain" which leaves whitespace? If so, see `_styles/citation.scss` to edit.

Perhaps you want to make each image just be a fixed width/height, not do any object-fitting, and maybe align them with the top of the citation cards (which will leave white space below them sometimes when the text content is long).

Perhaps you want to make every citation _card_ (the whole component, not just the image) a fixed height, but you have [some citations with very long descriptions and other content that makes them much taller than the rest](https://inspirelab-site.github.io/papers/?search=Investigated+the+interaction), so you'd have to decide what do to about that: truncate long text? get rid of tags/links? The Manubot catalog takes the approach of limiting the title/authors/extras to a max of 2 lines each to achieve this.

> The favicon still displays the website lab template even though I have replaced all files in ./images.

I am seeing the correct favicon for your site:

<img width="185" alt="Screenshot 2024-07-24 at 1 28 01 PM" src="https://github.com/user-attachments/assets/ccfcfd67-2fac-4bb9-a885-fcb385429327">

Try [hard-refreshing your browser](https://www.google.com/search?q=safari+hard+refresh). Unfortunately, browsers sometimes aggressively cache the favicon image, even though it is (supposed to be) small in size.

I noticed that [your `icon.png`](https://raw.githubusercontent.com/inspirelab-site/inspirelab-site.github.io/main/images/icon.png) image is pretty big. If you reduce it to 512 x 512 (or even further), it _might_ become small enough where browsers don't bother to cache it at all. This is just a guess though; completely dependent on the under-the-hood implementation details of each browser.

> The icon gets squeezed when narrowing the window size or browsing on iPhone/iPad.

This is a [bug](https://github.com/greenelab/lab-website-template/issues/269). In `_styles/header.scss` within `.logo {`, add `flex-shrink: 0;`.

**Comment by imnanxu** [comment]
Regarding the paper types, I did not see any [specified faded icons](https://greene-lab.gitbook.io/lab-website-template-docs/basics/citations#rich-details) shown on the published items with specified types. Could you please guide me on how to enable these icons to be visualized?

**Reply by vincerubinetti** [reply]
I'm not sure what you're asking. You are specifying the `type` for all of your regular sources:
https://github.com/inspirelab-site/inspirelab-site.github.io/blob/main/_data/sources.yaml

And they seem to be showing up:
https://inspirelab-site.github.io/papers/?search=+pmid%3A31217565#2019

<img height="100" src="https://github.com/user-attachments/assets/7e65b713-89aa-4d90-92ec-eac876d0db90" />
<img height="100" src="https://github.com/user-attachments/assets/711c7f26-d76f-4fb4-8beb-5885b90f19f7" />
